src/DynamicDataVNext/Kernel/KeyedChangeSet.cs

```diff
@@ -84,6 +130,54 @@ public static class KeyedChangeSet
         };
     }
 
+    /// <summary>
+    /// Creates a new <see cref="KeyedChangeSet{TKey, TItem}"/> representing the clearing of a keyed collection.
+    /// </summary>
+    /// <typeparam name="TKey">The type of the items' keys.</typeparam>
+    /// <typeparam name="TItem">The type of the items being removed.</typeparam>
+    /// <param name="items">The items being removed.</param>
+    /// <param name="keySelector">A selector for determining the key for each item being removed.</param>
+    /// <returns>A <see cref="KeyedChangeSet{TKey, TItem}"/> describing the clearing of the collection.</returns>
+    public static KeyedChangeSet<TKey, TItem> Clear<TKey, TItem>(
+        IEnumerable<TItem>  items,
+        Func<TItem, TKey>   keySelector)
+    {
+        if (!items.TryGetNonEnumeratedCount(out var itemsCount))
```

Comment:
Until your recent work, I did not even know about TryGetNonEnumeratedCount.  I can see for priming any collection this is super useful

Reply:
And easily polyfill-able for older target frameworks.